packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx

```diff
@@ -230,9 +236,12 @@ function ParamArrayFormItem({ param }: ParamProps) {
           </button>
         </div>
       ))}
-      <button className={styles.buttonThin} onClick={handleAddItem}>
-        Add item
-      </button>
+      {(param?.schema?.maxItems == null ||
+        items.length < param.schema.maxItems) && (
+        <button className={styles.buttonThin} onClick={handleAddItem}>
+          Add item
+        </button>
+      )}
```

Comment:
I decided to just hide the "Add item" button once we reach the `maxItems` limit here, but another approach would be to show a diabled button instead, like this:

```jsx
{(param?.schema?.maxItems == null || items.length < param.schema.maxItems) &&
  <button className={styles.buttonThin} onClick={handleAddItem}>
    Add item
  </button>
  ||
  <button className={styles.buttonThin} disabled>
    Add item
  </button>
}
```

I don't have a really strong view on it.

Reply:
Could we do disabled instead? I think that’s more accessibility friendly

Reply:
:+1: done in 71ac129